src/utils/beacon/BeaconPeers.tsx

```diff
@@ -52,64 +49,83 @@ export const BeaconPeers = () => {
     );
   }
 
-  return <PeersDisplay peerInfos={peersWithId} removePeer={removePeer} />;
+  return (
+    <Box>
+      {peersWithId.map(peerInfo => (
+        <PeersDisplay key={peerInfo.name} peerInfo={peerInfo} />
+      ))}
+    </Box>
+  );
 };
 
 /**
- * Component for displaying a list of connected dApps.
+ * Component for displaying list of connections for given dApp.
  *
- * Each {@link PeerRow} contains info about a single dApp & delete button.
+ * If there are no saved connections, one row for the dApp will still be displayed.
  *
- * @param peerInfos - peerInfo provided by beacon Api + computed dAppId.
- * @param removePeer - hook for deleting dApp connections.
+ * @param peerInfo - peerInfo provided by beacon Api + computed dAppId.
  */
-const PeersDisplay = ({
-  peerInfos,
-  removePeer,
-}: {
-  peerInfos: ExtendedPeerInfo[];
-  removePeer: (peer: ExtendedPeerInfo) => void;
-}) => (
-  <Box>
-    {peerInfos.map(peerInfo => (
+const PeersDisplay = ({ peerInfo }: { peerInfo: ExtendedPeerInfo }) => {
+  const getConnectionInfos = useGetConnectedAccounts();
+  const connectedAccounts = getConnectionInfos(peerInfo.senderId);
+
+  if (connectedAccounts.length === 0) {
```

Comment:
is this state possible?

Reply:
Yes. It's rare, but possible.
Connections without saved info will exit in case if connection was created before we added BeaconSlice.